From ea4c839b76a9c513fbe28b805b35b2a9681f70fd Mon Sep 17 00:00:00 2001
From: Konrad Weihmann <kweihmann@outlook.com>
Date: Sat, 15 Jun 2024 10:44:55 +0000
Subject: [PATCH] erubi: fall back on cgi/escape

also on NameError, which is caused by the either missing or
not fully available erb/escape native extension

Upstream-Status: Inappropriate [oe-specific]
Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>
---
 lib/erubi.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/erubi.rb b/lib/erubi.rb
index dd7db06..babad69 100644
--- a/lib/erubi.rb
+++ b/lib/erubi.rb
@@ -21,7 +21,7 @@ module Erubi
     require 'erb/escape'
     define_method(:h, ERB::Escape.instance_method(:html_escape))
   # :nocov:
-  rescue LoadError
+  rescue LoadError, NameError
     begin
       require 'cgi/escape'
       unless CGI.respond_to?(:escapeHTML) # work around for JRuby 9.1
-- 
2.34.1

